From 1d6ae5b7e8d0773ad80d96efcbe888a51863ed51 Mon Sep 17 00:00:00 2001
From: Bixuan Cui <cuibixuan@huawei.com>
Date: Fri, 3 Dec 2021 18:15:48 +0800
Subject: [PATCH] suspend: export cpu_suspend/cpu_resume/psci_ops
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 1d6ae5b7e8d0773ad80d96efcbe888a51863ed51


ascend inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4K2U5
CVE: NA

-------------------------------------------------

Export cpu_suspend/cpu_resume/psci_ops for lowpower driver.

Signed-off-by: Bixuan Cui <cuibixuan@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/suspend.c  | 2 ++
 drivers/firmware/psci/psci.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm64/kernel/suspend.c b/arch/arm64/kernel/suspend.c
index 9f8cdeccd1ba..acef0422ab2e 100644
--- a/arch/arm64/kernel/suspend.c
+++ b/arch/arm64/kernel/suspend.c
@@ -134,6 +134,8 @@ int cpu_suspend(unsigned long arg, int (*fn)(unsigned long))
 
 	return ret;
 }
+EXPORT_SYMBOL_GPL(cpu_suspend);
+EXPORT_SYMBOL_GPL(cpu_resume);
 
 static int __init cpu_suspend_init(void)
 {
diff --git a/drivers/firmware/psci/psci.c b/drivers/firmware/psci/psci.c
index 00af99b6f97c..151d00898cab 100644
--- a/drivers/firmware/psci/psci.c
+++ b/drivers/firmware/psci/psci.c
@@ -47,6 +47,8 @@
  */
 static int resident_cpu = -1;
 struct psci_operations psci_ops;
+EXPORT_SYMBOL(psci_ops);
+
 static enum arm_smccc_conduit psci_conduit = SMCCC_CONDUIT_NONE;
 
 bool psci_tos_resident_on(int cpu)
-- 
2.26.2

